.envrc

```diff
@@ -124,13 +124,13 @@ make setup-git-config
 # we're enforcing that virtualenv be in .venv{,3}, since future tooling e.g.
 # venv-update will rely on this.
 venv_name=".venv"
-python_version="python2.7"
+python_version="python3.6"
```

Comment:
It would be nice to check if they already have a python2 venv setup in their `.venv` folder, and if they do switch `python_version` back to `python2.7` and then drop a huge console warning indicating that they should upgrade

Reply:
I have not tested via `direnv`, however, via `make bootstrap` it works:

```shell
❯ pyenv local 2.7.16
❯ python -m virtualenv .venv
❯ make bootstrap
cat: Remove /Users/armenzg/code/sentry/.venv and try again since the Python version installed should be 3.6.: No such file or directory
make: *** [ensure-venv] Error 1
```

Do you want me to put more effort to address the path via `direnv`?

Reply:
Since this would currently break immediately for direnv users, I would suggest it yes.

Basically I think everything should continue to work with existing .venv directories until a user consciously does the make dance to switch

Reply:
The experience for `direnv` users as of last night's code:
```
git checkout master
# Follow the prompts of direnv
python2.7 -m virtualenv .venv
source .venv/bin/activate
make install-py-dev
make setup-git
# Let's try the new code
git checkout python3_by_default
❯ direnv allow
direnv: loading ~/code/sentry/.envrc
direnv: Configuring git...
direnv: Activating virtualenv...
direnv FATAL: For some reason, the virtualenv isn't Python 3.6.
```

After [70a811dc159d01c1effcfa0e979efd6b0f56f715](https://github.com/getsentry/sentry/pull/22284/commits/70a811dc159d01c1effcfa0e979efd6b0f56f715):
```
❯ direnv allow
<trimmed output>
direnv: We are moving to Python 3, recreate your venv if you want to start using it
direnv FATAL: If you want to keep using your Python 2 venv please set export SENTRY_PYTHON2=1
❯ export SENTRY_PYTHON2
❯ direnv allow
<trimmed output>
direnv: Running in EXPERIMENTAL Python 2 mode...
<trimmed output>
direnv: SUCCESS!
```

Reply:
Does it use the .venv directory in this case, or do they have to setup a .venv2 env?

Reply:
@EvanPurkhiser If they have a `.venv` with Python 2 it will give them a warning and will ask them to use `SENTRY_PYTHON2=1` to keep using it.

If they want to have both environments (`.venv` with Python 3 and `.venv2` with Python 2) they can do so.

I've pushed one more commit to say `.venv2` for the direnv instructions.